File: emulinker/src/main/java/org/emulinker/kaillera/controller/v086/protocol/GameChat_Notification.java
package org.emulinker.kaillera.controller.v086.protocol;

import com.google.auto.value.AutoValue;
import org.emulinker.kaillera.controller.messaging.MessageFormatException;

@AutoValue
public abstract class GameChat_Notification extends GameChat {
  private static final String DESC = "In-Game Chat Notification";

  public static AutoValue_GameChat_Notification create(
      int messageNumber, String username, String message) throws MessageFormatException {
    V086Message.validateMessageNumber(messageNumber, DESC);

    return new AutoValue_GameChat_Notification(messageNumber, ID, DESC, username, message);
  }
}

File: emulinker/src/main/java/org/emulinker/kaillera/controller/v086/protocol/GameChat_Notification.kt
package org.emulinker.kaillera.controller.v086.protocol

import org.emulinker.kaillera.controller.messaging.MessageFormatException
import org.emulinker.kaillera.controller.v086.protocol.V086Message.Companion.validateMessageNumber

data class GameChat_Notification
    @Throws(MessageFormatException::class)
    constructor(
        override val messageNumber: Int, override val username: String, override val message: String
    ) : GameChat() {

  override val messageId = ID

  init {
    validateMessageNumber(messageNumber)
  }
}
